synchronizer/local.go

```diff
@@ -63,13 +94,27 @@ func (s *Local) RefreshRates() (time.Duration, time.Duration) {
 
 // SynchronizeSplits syncs splits
 func (s *Local) SynchronizeSplits(till *int64) error {
-	_, err := s.workers.SplitFetcher.SynchronizeSplits(nil)
+	result, err := s.workers.SplitFetcher.SynchronizeSplits(till)
+	for _, segment := range s.filterCachedLocalSegments(result.ReferencedSegments) {
```

Comment:
you need to take into account here that maybe there is no segmentDirectory attached to avoid this logic

Reply:
Added check if s.workers.SegementFetcher is nil.